rmgpy/kinetics/arrhenius.pyx

```diff
@@ -597,57 +597,80 @@ cdef class ArrheniusBM(KineticsModel):
             w0s = get_w0s(recipe, rxns)
             w0 = sum(w0s) / len(w0s)
 
-        # define optimization function
-        def kfcn(xs, lnA, n, E0):
-            out = []
-            for x in xs:
-                T = x[0]
-                dHrxn = x[1]
-                if dHrxn < -4 * E0:
-                    Ea = 0.0
-                elif dHrxn > 4 * E0:
-                    Ea = dHrxn
-                else:
-                    Vp = 2 * w0 * (2 * w0 + 2 * E0) / (2 * w0 - 2 * E0)
-                    Ea = (w0 + dHrxn / 2.0) * (Vp - 2 * w0 + dHrxn) ** 2 / (Vp ** 2 - (2 * w0) ** 2 + dHrxn ** 2)
-
-                out.append(lnA + np.log(T ** n * np.exp(-Ea / (8.314 * T))))
-            return out
-
-        # get (T,dHrxn(T)) -> (Ln(k) mappings
-        xdata = []
-        ydata = []
-        sigmas = []
-        for rxn in rxns:
-            # approximately correct the overall uncertainties to std deviations
-            s = rank_accuracy_map[rxn.rank].value_si/2.0
-            for T in Ts:
-                xdata.append([T, rxn.get_enthalpy_of_reaction(T)])
-                ydata.append(np.log(rxn.get_rate_coefficient(T)))
-
-                sigmas.append(s / (8.314 * T))
-
-        xdata = np.array(xdata)
-        ydata = np.array(ydata)
-
-        # fit parameters
-        boo = True
-        xtol = 1e-8
-        ftol = 1e-8
-        while boo:
-            boo = False
-            try:
-                params = curve_fit(kfcn, xdata, ydata, sigma=sigmas, p0=[1.0, 1.0, w0 / 10.0], xtol=xtol, ftol=ftol)
-            except RuntimeError:
-                if xtol < 1.0:
-                    boo = True
-                    xtol *= 10.0
-                    ftol *= 10.0
-                else:
-                    raise ValueError("Could not fit BM arrhenius to reactions with xtol<1.0")
-
-        lnA, n, E0 = params[0].tolist()
-        A = np.exp(lnA)
+        if len(rxns) == 1:
+            T = 1000.0
+            rxn = rxns[0]
+            dHrxn = rxn.get_enthalpy_of_reaction(T)
+            A = rxn.kinetics.A.value_si
+            n = rxn.kinetics.n.value_si
+            Ea = rxn.kinetics.Ea.value_si
+            
+            def kfcn(E0):
```

Comment:
What happens when abs(dHrxn) > 4*E0? Can we just guess E0 = w/10.0 if that's the case?

Reply:
> What happens when abs(dHrxn) > 4*E0? Can we just guess E0 = w/10.0 if that's the case?

Thanks for the comment @mjohnson541 ! I changed the code accordingly.